apps/meteor/client/views/room/contextualBar/RoomMembers/AddUsers/AddUsersWithData.tsx

```diff
@@ -52,13 +65,44 @@ const AddUsersWithData = ({ rid, onClickBack, reload }: AddUsersWithDataProps):
 			dispatchToastMessage({ type: 'error', message: error as Error });
 		}
 	});
+
+	const handleMatrixValidation = useMutableCallback(async () => {
+		try {
+			const matrixIds = users.filter((user) => user.startsWith('@'));
+
+			const matrixIdsVerificationPromises = matrixIds.map((matrixId) =>
+				dispatchVerifyEndpoint({ matrixId } as FederationVerifyMatrixIdProps),
+			);
+
+			const matrixIdsVerificationPromiseResponse = await Promise.allSettled(matrixIdsVerificationPromises);
+			const matrixIdsVerificationFulfilledResults = matrixIdsVerificationPromiseResponse.filter(
+				(res) => res.status === 'fulfilled',
+			) as PromiseFulfilledResult<{ result: string }>[];
+
+			for (let i = 0; i < matrixIds.length; i++) {
```

Comment:
maybe use `.map(() => ...)` for this? Also, you're using a `filter` for `matrixIdsVerificationFulfilledResults` which can cause the array to be smaller than `matrixIds`. This by itself can cause 2 problems:
1 - `matrixIdsVerificationFulfilledResults[i];` can be undefined if `matrixIds` is bigger;
2 - the ID array position won't match with the response array.